execution/gethexec/timeboost_sequencer.go

```diff
@@ -0,0 +1,549 @@
+			return false
+		default:
+		}
+
+		if done {
+			// We have added as many transactions as we can
+			break
+		}
+
+		if queueItem.txSize > s.config().MaxTxDataSize {
+			// This tx is too large
+			// Even if its a priority item this should be skipped,
+			// TODO: return the error to the user here
+			log.Warn("timeboost transaction is too large", "txSize", queueItem.txSize, "maxTxDataSize", s.config().MaxTxDataSize, "hash", queueItem.tx.Hash().Hex())
+			continue
+		}
+
+		if arbmath.BigLessThan(queueItem.tx.GasFeeCap(), lastBlock.BaseFee) {
+			// This tx is too low gas fee
+			// TODO: return the error to the user here
+			log.Warn("timeboost transaction has too low gas fee", "txSize", queueItem.txSize, "gasFeeCap", queueItem.tx.GasFeeCap(), "baseFee", lastBlock.BaseFee, "hash", queueItem.tx.Hash().Hex())
+			continue
+		}
+
+		if totalBlockSize+queueItem.txSize > s.config().MaxTxDataSize {
+			// This tx would be too large to add to this batch
+			log.Info("timeboost transaction is too large, adding to retry queue", "txSize", queueItem.txSize, "maxTxDataSize", s.config().MaxTxDataSize, "hash", queueItem.tx.Hash().Hex())
+			s.txRetryQueue.Push(queueItem)
+			// End the batch here to put this tx in the next one
+			break
+		}
+		totalBlockSize += queueItem.txSize
+		queueItems = append(queueItems, queueItem)
+	}
+
+	s.nonceCache.Resize(config.NonceCacheSize)
+	s.nonceCache.BeginNewBlock()
+	queueItems = s.precheckNonces(queueItems)
+	txes := make([]*types.Transaction, len(queueItems))
+	hooks := s.makeSequencingHooks()
+	hooks.ConditionalOptionsForTx = make([]*arbitrum_types.ConditionalOptions, len(queueItems))
+	totalBlockSize = 0
+	for i, queueItem := range queueItems {
+		txes[i] = queueItem.tx
+		totalBlockSize = arbmath.SaturatingAdd(totalBlockSize, queueItem.txSize)
+		hooks.ConditionalOptionsForTx[i] = queueItem.options
+	}
+	if totalBlockSize > config.MaxTxDataSize {
+		for _, queueItem := range queueItems {
+			s.txRetryQueue.Push(queueItem)
+		}
+		log.Error(
+			"put too many transactions in a block",
+			"numTxes", len(queueItems),
+			"totalBlockSize", totalBlockSize,
+			"maxTxDataSize", config.MaxTxDataSize,
+		)
+		return false
+	}
+	if len(queueItems) == 0 {
+		return true
+	}
+
+	timestamp := queueItems[0].consensusTimestamp
+	header, err := s.l1Reader.LatestFinalizedBlockHeader(ctx)
+	if err != nil {
+		log.Error("failed to get latest finalized block header", "err", err)
+		return false
+	}
+	// finalized l1 block <= consensus timestamp - parent chain finalization time
+	l1Block, err := s.getL1BlockNumber(ctx, header.Number.Int64(), header.Time)
+	if err != nil {
+		return false
+	}
+
+	l1IncomingMessageHeader := &arbostypes.L1IncomingMessageHeader{
+		Kind:        arbostypes.L1MessageType_L2Message,
+		Poster:      l1pricing.BatchPosterAddress,
+		BlockNumber: l1Block.NumberU64(),
+		Timestamp:   timestamp,
+		RequestId:   nil,
+		L1BaseFee:   nil,
+	}
+
+	start := time.Now()
+	var (
+		block *types.Block
+	)
```

Comment:
Why do it in a scoped block like this?

Reply:
I will update it, not sure why I did it lol